driver/network_driver.go

```diff
@@ -252,8 +290,12 @@ func (d NetworkDriver) Join(request *network.JoinRequest) (*network.JoinResponse
 		return nil, err
 	}
 
+	mac, err := d.GetEndpointMAC(request.EndpointID)
+	if err != nil {
+		return nil, err
+	}
 	// libnetwork doesn't set the MAC address properly, so set it here.
```

Comment:
Can you double check that his comment is still actually true? I'm a little uneasy that we're introducing an extra trip to the datastore and I would be much happier if we didn't have to.